eixampledb-api/src/main/java/com/eixampledb/api/MainController.java

```diff
@@ -27,54 +25,36 @@
     public ResponseEntity<String> get(@PathVariable("key") String key) {
         GetResponse getResponse = eixampledb.get(new GetRequest(key));
         if (getResponse.isSuccess()) {
-            return ResponseEntity.ok(getResponse.getEntry().get().getValue());
+            return ResponseEntity.ok(String.valueOf(getResponse.getEntry().get().getValue()));
         } else {
             return ResponseEntity.notFound().build();
         }
     }
 
     @RequestMapping(path = "/{key}", method = RequestMethod.POST)
-    public ResponseEntity set(@PathVariable("key") String key, @RequestBody String value, @RequestHeader Map<String,String> header) {
-        String type = header.get("type");
-        int t;
-        if (type == null){
-            type = "";
-        }
-        if (type.equals("NUM")){
-            t = core.NUM_TYPE;
-        }else if (type.equals("STR")){
-            t = core.STRING;
-        }
-        else{
-            t = core.STRING;
-        }
-        eixampledb.set(new SetRequest(key, value, t));
+    public ResponseEntity set(@PathVariable("key") String key,
+                              @RequestBody String value,
+                              @RequestHeader(name = "type", defaultValue = "STR", required = false) ValueType valueType) {
+        eixampledb.set(new SetRequest(key, value, valueType));
         return ResponseEntity.ok().build();
     }
 
     @RequestMapping(path = "/{key}", method = RequestMethod.PUT)
-    public ResponseEntity operation(@PathVariable("key") String key, @RequestHeader("op") String operation) {
-
-        GetResponse getResponse = eixampledb.get(new GetRequest(key));
```

Comment:
That get-set behaviour implies that the increment and decrement operations are not atomic. All operations have to be atomic. Therefore to be atomic it has to be delegated to EixampleDb implementation.